### DISCUSSION TITLE
Issue with deploying contract on ZKSync (but compiling is ok)

### DISCUSSION THREAD

**Discussion by JohannCali** [Cyfrin/foundry-full-course-cu#2832] [General]
![Capture d'écran 2024-11-09 130051](https://github.com/user-attachments/assets/cfc2688a-7a9f-48bd-8e4b-182fb7046919)

I tried to fix it alone but i didn't succeed.

**Comment by EngrPips** [comment]
Hello @JohannCali, Did you put your files in a `contracts` folder?

**Reply by JohannCali** [reply]
No, should I ?

**Reply by EngrPips** [reply]
Yeah, Do and let's see if that works

**Reply by ghost** [reply]
I have the same issue. Deploying contract on zksync creates additional, nested artifacts folder with additional contracts folder containing duplicated SimpleStorage.sol file inside of a original artifacts folder. Environment selected as wallet. Lesson 16. Zksync plugin, Solidity smart contract development.

**Reply by EngrPips** [reply]
I am not sure how that happened, maybe clear your workspace and retry @nikola755

**Reply by JohannCali** [reply]
It works ! Thank you for your answer !

**Reply by EngrPips** [reply]
Awesome.

**Reply by KAravind2005** [reply]
thanks a lot brother. I too gone through this issue. yes It works. Make sure the file in the contracts folder. Thats it..

**Reply by EngrPips** [reply]
Interesting

**Comment by ghost** [comment]
If you have previously manually deleted folders inside workspace, delete worskpace, and reopen Remix ide. After that write your code inside contracts folder and it should work

**Comment by EngrPips** [comment]
Good heads-up
